### PR TITLE
Fix performance bug with regional authority detection

### PR DESCRIPTION
**Pull request by jo-arroyo**
This PR addresses a performance bug where the Azure region is still being auto-detected even when the Azure region is configured, for telemetry purposes. Refactored function will use Azure region where configured for performance advantages. This fix passes all existing unit tests for regional discovery and user provided region overrides.

Note: Although not all telemetry outcomes will be used after refactoring, `RegionDiscoveryOutcomes` constant remains the same to reflect spec-defined telemetry data points.

### CODE REVIEW

**Review by peterzenz — APPROVED** [review]

**Review by tnorling — APPROVED** [review]

File: change/@azure-msal-common-3cdc9d22-67a6-41c7-8d7c-6b42b43c8106.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (sameerag) [inline change/@azure-msal-common-3cdc9d22-67a6-41c7-8d7c-6b42b43c8106.json:1]: Do we need this in v2 also?

File: lib/msal-common/src/authority/Authority.ts [hunk 2]
```diff
@@ -579,47 +580,38 @@ export class Authority {
...
-            if (autodetectedRegionName) {
-                this.regionDiscoveryMetadata.region_outcome =
-                    this.authorityOptions.azureRegionConfiguration
-                        ?.azureRegion === autodetectedRegionName
-                        ? RegionDiscoveryOutcomes.CONFIGURED_MATCHES_DETECTED
-                        : RegionDiscoveryOutcomes.CONFIGURED_NOT_DETECTED;
-            } else {
-                this.regionDiscoveryMetadata.region_outcome =
-                    RegionDiscoveryOutcomes.CONFIGURED_NO_AUTO_DETECTION;
+        if (userConfiguredAzureRegion) {
```
Comment (sameerag) [inline lib/msal-common/src/authority/Authority.ts:585]:
```suggestion
        if (userConfiguredAzureRegion && userConfiguredAzureRegion !== Constants.AZURE_REGION_AUTO_DISCOVER_FLAG) {
```

File: lib/msal-common/src/authority/Authority.ts [hunk 3]
```diff
@@ -579,47 +580,38 @@ export class Authority {
...
+            if (userConfiguredAzureRegion !== Constants.AZURE_REGION_AUTO_DISCOVER_FLAG) {
+                this.regionDiscoveryMetadata.region_outcome = RegionDiscoveryOutcomes.CONFIGURED_NO_AUTO_DETECTION;
+                this.regionDiscoveryMetadata.region_used = userConfiguredAzureRegion;
```
Comment (sameerag) [inline lib/msal-common/src/authority/Authority.ts:588]: Do we cover for user passing an `invalid region`? How does the e2e flow work there? Does the service throw?

**Review by sameerag — APPROVED** [review]
Left few comments, approved.

File: change/@azure-msal-common-3cdc9d22-67a6-41c7-8d7c-6b42b43c8106.json (hunk 1 above)
Comment (jo-arroyo) [reply change/@azure-msal-common-3cdc9d22-67a6-41c7-8d7c-6b42b43c8106.json:1]: I will open a separate PR for this change in v2.

File: lib/msal-common/src/authority/Authority.ts (hunk 3 above)
Comment (jo-arroyo) [reply lib/msal-common/src/authority/Authority.ts:588]: According to the spec it seems we just pass on the user input region, and it is up to the service to throw if the region is invalid.

**Pull request merged**